osu.Game/Screens/Edit/Compose/Components/Timeline/SamplePointPiece.cs

```diff
@@ -43,15 +49,35 @@ public SamplePointPiece(HitObject hitObject)
 
         protected override Color4 GetRepresentingColour(OsuColour colours) => AlternativeColor ? colours.Pink2 : colours.Pink1;
 
+        protected virtual double GetTime() => HitObject is IHasRepeats r ? HitObject.StartTime + r.Duration / r.SpanCount() / 2 : HitObject.StartTime;
+
         [BackgroundDependencyLoader]
         private void load()
         {
             HitObject.DefaultsApplied += _ => updateText();
             updateText();
+
+            if (editor != null)
+                editor.ShowSampleEditPopoverRequested += onShowSampleEditPopoverRequested;
+        }
+
+        protected override void Dispose(bool isDisposing)
+        {
+            base.Dispose(isDisposing);
+
+            if (editor != null)
+                editor.ShowSampleEditPopoverRequested -= onShowSampleEditPopoverRequested;
+        }
+
+        private void onShowSampleEditPopoverRequested(double time)
+        {
+            if (time == GetTime())
+                this.ShowPopover();
         }
 
         protected override bool OnClick(ClickEvent e)
         {
+            editorClock?.SeekSmoothlyTo(GetTime());
```

Comment:
I'm still not sure about this UX. Maybe best to only trigger this seek when using the seek keys?

Results in big reach:

https://github.com/user-attachments/assets/9277306a-ae8e-44e9-96d0-1d50b94aff76

Reply:
I see your point. I'm not sure what you mean by the seek keys. Should it only seek when you click it while Control is held? 

I think it might be best to trigger the seek on a right click maybe, or some other mouse button, because I expect the mapper to be already using the hotkeys for assigning hitsounds when using this feature, so adding modifiers on click makes the flow more complicated.

Reply:
I would've liked to keep the right-click navigation but I assume that's a peppy-no-go

Reply:
I dunno. It's too many random UX behaviours. I want to avoid this unless it's well defined (ie. something that matches other apps, or something implemented everywhere in lazer).

It's a no-go for this PR.